### PR TITLE
Remove gnulib

### PR DESCRIPTION
**Pull request by clumens**
This changes build requirements in such a way that this will not be backported.

### CODE REVIEW

File: lib/common/tests/digest/crm_md5sum_test.c [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+#include <crm/common/unittest_internal.h>
+
+static void
+basic_usage_test(void **state) {
+    char *result = crm_md5sum("abcdefghijklmnopqrstuvwxyz");
+
+    assert_string_equal(result, "c3fcd3d76192e4007dfb496cca67e13b");
+    free(result);
+}
```
Comment (kgaillot) [inline lib/common/tests/digest/crm_md5sum_test.c:20]: worth at least adding a test for NULL arg

File: configure.ac [hunk 2]
```diff
@@ -1132,6 +1125,7 @@ REQUIRE_HEADER([fcntl.h])
 REQUIRE_HEADER([float.h])
 REQUIRE_HEADER([glib.h])
 REQUIRE_HEADER([grp.h])
+REQUIRE_HEADER([gnutls/gnutls.h])
```
Comment (kgaillot) [inline configure.ac:1128]: don't need this since `PKG_CHECK_MODULES` will require it

File: configure.ac [hunk 3]
```diff
@@ -1678,27 +1672,10 @@ dnl ========================================================================
...
-          PC_NAME_GNUTLS="gnutls"
-          PCMK_FEATURES="$PCMK_FEATURES remote"
-      ]
-)
-AC_SUBST([PC_NAME_GNUTLS])
-AM_CONDITIONAL([BUILD_REMOTE], [test $with_gnutls -ne $DISABLED])
+PKG_CHECK_MODULES(GNUTLS, [gnutls >= 2.10.0],
+                  [CPPFLAGS="${CPPFLAGS} ${GNUTLS_CFLAGS}"
+                   LIBS="${LIBS} ${GNUTLS_LIBS}"])
+PCMK_FEATURES="$PCMK_FEATURES remote"
```
Comment (kgaillot) [inline configure.ac:1678]: we can drop the FEATURES line (we only use it for optional stuff)

File: daemons/execd/Makefile.am [hunk 4]
```diff
@@ -13,6 +13,7 @@ include $(top_srcdir)/mk/man.mk
 halibdir		= $(CRM_DAEMON_DIR)
 
 halib_PROGRAMS		= pacemaker-execd \
+			  pacemaker-remoted \
```
Comment (kgaillot) [inline daemons/execd/Makefile.am:16]: don't need this (it goes in sbin)

File: daemons/execd/Makefile.am [hunk 5]
```diff
@@ -27,7 +28,6 @@ pacemaker_execd_SOURCES	= pacemaker-execd.c 	\
 			  execd_commands.c 	\
 			  execd_alerts.c
 
-if BUILD_REMOTE
```
Comment (kgaillot) [inline daemons/execd/Makefile.am:30]: also drop all the C checks for `HAVE_GNUTLS_GNUTLS_H`

File: lib/common/digest.c [hunk 6]
```diff
@@ -250,28 +250,35 @@ pcmk__xa_filterable(const char *name)
...
+    unsigned int dlen = gnutls_hash_get_len(GNUTLS_DIG_MD5);
+    unsigned char *raw_digest = NULL;
     char *digest = NULL;
-    unsigned char raw_digest[MD5_DIGEST_SIZE];
+    int rc = 0;
 
-    if (buffer == NULL) {
-        buffer = "";
+    if (dlen == 0) {
+        return NULL;
```
Comment (kgaillot) [inline lib/common/digest.c:259]: handle NULL argument too (returning NULL is fine since we ensure non-NULL internally)

**Review by kgaillot — COMMENTED** [review]
beautiful

File: lib/common/digest.c (hunk 6 above)
Comment (clumens) [reply lib/common/digest.c:259]: Fixed.

File: daemons/execd/Makefile.am (hunk 5 above)
Comment (clumens) [reply daemons/execd/Makefile.am:30]: Fixed.  Note that this removes guards from around members of `pcmk__remote_s` and a couple flag enums, among other spots.  I assume this is not going to be a problem because we won't be supported clusters consisting of v3.0 and v2.x at the same time.

**Review by kgaillot — COMMENTED** [review]
looks great

**Review by kgaillot — APPROVED** [review]
